packages/backend/src/QueryEngine.ts

```diff
@@ -131,7 +148,34 @@ export class QueryEngine<DB> {
                 connectionString: config.url,
                 max: 10,
             });
-        this.middlewares = config.middlewares ?? [];
+        this.middlewares = [
+            ...(config.middlewares ?? []),
+            middleware<AnyQuery, AnyContext>({
```

Comment:
let's move the declaration of this middleware to its own file

Reply:
The implementation is also very odd to me. I need to think of a way to simplify it.

Reply:
Ok I see what's going on: the middleware is not running recursively over the queries, and so you need to it manually here.

Don't you think we should move the middleware to iterate over queries in the query tree recursively?